src/Platforms/Gcc/UtestPlatform.cpp

```diff
@@ -76,29 +76,33 @@ static int PlatformSpecificWaitPidImplementation(int, int*, int)
 
 static void GccPlatformSpecificRunTestInASeperateProcess(UtestShell* shell, TestPlugin* plugin, TestResult* result)
 {
-    pid_t cpid, w;
+    const pid_t syscallError = -1;
+    pid_t cpid;
+    pid_t w;
     int status;
 
     cpid = PlatformSpecificFork();
 
-    if (cpid == -1) {
+    if (cpid == syscallError) {
         result->addFailure(TestFailure(shell, "Call to fork() failed"));
         return;
     }
 
     if (cpid == 0) {            /* Code executed by child */
-        shell->runOneTestInCurrentProcess(plugin, *result);   // LCOV_EXCL_LINE
-        _exit(result->getFailureCount());                     // LCOV_EXCL_LINE
+        const int initialFailureCount = result->getFailureCount(); // LCOV_EXCL_LINE
+        shell->runOneTestInCurrentProcess(plugin, *result);        // LCOV_EXCL_LINE
+        _exit(initialFailureCount < result->getFailureCount());    // LCOV_EXCL_LINE
     } else {                    /* Code executed by parent */
+        result->countRun();
```

Comment:
Here I assume that the test is "run" only if `fork()` returns success to parent (as child is not run).
Alternatively, we can count in caller function like proposed in #1210. Or we can count in line 104 below, i.e. if `waitpid()` also passes.
I'm open if this needs to be changed.

Reply:
Seems that proposal would be better because then there is just one count, right?

Any trade-offs to be made here?

Reply:
I think, it is so. If we consider that "attempt to run"is "run" (i.e. failure to fort means that the test is not actually even started), then having it in one place is better.
I'll do it in one place then (unless I find strange things in "run one test" function

Reply:
Ack

Reply:
done